zio-http/src/main/scala/zhttp/socket/Socket.scala

```diff
@@ -16,10 +16,17 @@ sealed trait Socket[-R, +E, -A, +B] { self =>
     case FOrElse(sa, sb)             => sa(a) <> sb(a)
     case FMerge(sa, sb)              => sa(a) merge sb(a)
     case Succeed(a)                  => ZStream.succeed(a)
+    case Provide(m, r)               => m(a).asInstanceOf[ZStream[R, E, B]].provide(r.asInstanceOf[R])
```

Comment:
```suggestion
    case Provide(sa, r)               => sa(a).asInstanceOf[ZStream[R, E, B]].provide(r.asInstanceOf[R])
```

Reply:
@tusharmath  How about we keep it just `s`? Since  `sa` could mean a SocketApp.